### DISCUSSION TITLE
AIO Nextcloud in Kubernetes is Faster and Better than the Kubernetes Native Nextcloud

### DISCUSSION THREAD

**Discussion by denppa** [nextcloud/all-in-one#5619] [Questions] [answered]
Also raised this discussion on the k8s native helm chart repository, but no one seemed to notice. Upon testing, the AIO deployed in k8s is much faster than the native k8s helm chart.

Maybe NC should consider migrating the build process for AIO images to the k8s native side, because the performance is much, much, much better, when both helm releases are installed on the same machine under the same network environment.

The only thing that is slightly missing would be the more professional organization of helm values.yml, currently they feel like environment variables that mimic docker instead of following k8s structure.

Thank you for making this releasing actually perform normally. Loading a page takes ~3 seconds whereas the k8s native one is around 6s on average, and could stretch to 12 seconds at times.

**Comment by szaimen** [comment, marked as answer]
> Upon testing, the AIO deployed in k8s is much faster than the native k8s helm chart.
>
> Thank you for making this releasing actually perform normally. Loading a page takes ~3 seconds whereas the k8s native one is around 6s on average, and could stretch to 12 seconds at times.

Hi, thanks for this positive feedback! :)



> The only thing that is slightly missing would be the more professional organization of helm values.yml, currently they feel like environment variables that mimic docker instead of following k8s structure.

Yes, this is indeed exactly what we are doing in order to keep all the different deployment methods in sync. See https://github.com/nextcloud/all-in-one/blob/main/nextcloud-aio-helm-chart/update-helm.sh. So we are not going to adjust their naming because this would mean to split the helm chart out of AIO which would potentially introduce issues because it would need to be kept in sync manually.

**Comment by denppa** [comment]
Makes sense. However, do you know why the AIO chart performs better than the native chart? What is your special sauce? This way I can make a PR on the native k8s images.

**Reply by szaimen** [reply]
The aio images are tuned in various ways in order to make sure that it performs well

**Comment by PatrickHuetter** [comment]
@szaimen I'm also running my own nextcloud helm chart since years using the official nextcloud docker image and adding all dependencies as separate deployments/pods as Chart dependencies. But also here, having really powerful hardware and no hardware limits the official nextcloud image performs very bad in comparism with nextcloud AIO. That's why i'm thinking to switch to another deployment method / helm chart. What would be the best deployment way currently? Which helm chart should i choose? I also have to migrate my existing nextcloud instance to the new one then.

**Reply by szaimen** [reply]
Hi,

> What would be the best deployment way currently?

I would say the standard Nextcloud AIO deployment: https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-use-this

> Which helm chart should i choose?

I would recommend the Nextcloud AIO Helm-chart: https://github.com/nextcloud/all-in-one/tree/main/nextcloud-aio-helm-chart

**Reply by PatrickHuetter** [reply]
@szaimen when i see the pvc definitions i see that there are no good patterns. It's also not possible to define existingClaim. :( But if it works i'll have to live with that. There are no other alternatives currently, right?

Why doesn't Nextcloud optimize the standard Nextcloud Image just by default?

**Reply by szaimen** [reply]
> Why doesn't Nextcloud optimize the standard Nextcloud Image just by default?

Because it is a community maintained image - so the community is responsible for that. 

I am personally employed by the company and as main maintainer of AIO over the last years of course made sure that it performs well.

**Reply by vikram0015** [reply]
Hi,

Can you share me the steps to setup aio in kubernetes. The method i am trying to do throws database issue.

**Reply by DerKnerd** [reply]
@szaimen is there a proper migration path from plain installation in Kubernetes to All-in-One. Or ould you share some of the optimizations you guys did? Then I would tune the image myself :slightly_smiling_face:
